lib/bulbs-elements/util/in-view-monitor.js

```diff
@@ -96,6 +100,11 @@ const maybeEmitEvent = (monitoredItem) => {
     let event = new CustomEvent('exitviewport');
     monitoredItem.element.dispatchEvent(event);
   }
+
+  if(!inView && approachingViewport) {
+    let event = new CustomEvent('approachingviewport');
```

Comment:
I agree it doesn't need to be taken on here.  As for a good namespace, even if it was something simple as `bulbs-approaching-viewport`